Sources/SwiftDocC/Infrastructure/Symbol Graph/SymbolGraphLoader.swift

```diff
@@ -97,6 +97,18 @@ struct SymbolGraphLoader {
                 
                 // Store the decoded graph in `loadedGraphs`
                 loadingLock.sync {
+                    // Track the operating system platforms found in the symbol graphs of this module.
+                    if symbolsByPlatformRegisteredPerModule[moduleName] == nil {
+                        symbolsByPlatformRegisteredPerModule[moduleName] = [:]
+                    }
+                    let modulePlatform = symbolGraph.module.platform.name ?? ""
+                    symbolGraph.symbols.values.forEach { symbol in
+                        if symbolsByPlatformRegisteredPerModule[moduleName]?[modulePlatform] == nil {
+                            symbolsByPlatformRegisteredPerModule[moduleName]?[modulePlatform] = []
+                        }
+                        symbolsByPlatformRegisteredPerModule[moduleName]?[modulePlatform]?.append(symbol.identifier)
+                    }
```

Comment:
FYI: You can use default values in subscripts to avoid checking for `nil` here. 

Instead of 
```swift
if symbolsByPlatformRegisteredPerModule[moduleName] == nil {
    symbolsByPlatformRegisteredPerModule[moduleName] = [:]
}
```
you could use 
```swift
symbolsByPlatformRegisteredPerModule[moduleName, default: [:]]
```

Applying the same pattern to the indexed subscripts means that all of this can be written as 
```suggestion
                    for symbol in symbolGraphs.symbols {
                        symbolsByPlatformRegisteredPerModule[moduleName, default: [:]][modulePlatform, default: []].append(symbol.identifier)
                    }
```